iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1766,6 +1771,55 @@ private void updateStatus(IotHubConnectionStatus newConnectionStatus, IotHubConn
         else if (newConnectionStatus == IotHubConnectionStatus.DISCONNECTED)
         {
             correlationCallbackCleanupThread.interrupt();
+            finalizeMultiplexedDevicesMessages(deviceId);
```

Comment:
Now when a multiplexed device reaches the DISCONNECTED state, we look for any of its messages in the outgoing queues and cancel them. This should address #1734